### ISSUE TITLE
Cannot have multiple options starting with same letter

### ISSUE THREAD

**Issue by druchoo** [moigagoo/cliar#1]
This code

```python
#!/usr/bin/env python3

from cliar import Cliar


class Cli(Cliar):
    def _root(self, url="", username=""):
        pass


if __name__ == "__main__":
    cli = Cli()
    cli.parse()
```
Returns

> Traceback (most recent call last):
>   File "./cli.py", line 12, in <module>
>     cli = Cli()
>   File "/usr/local/lib/python3.7/site-packages/cliar/cliar.py", line 126, in __init__
>     self._register_root_args()
>   File "/usr/local/lib/python3.7/site-packages/cliar/cliar.py", line 147, in _register_root_args
>     self._register_arg(self._parser, arg_name, arg_data)
>   File "/usr/local/lib/python3.7/site-packages/cliar/cliar.py", line 188, in _register_arg
>     help=arg_data.help
>   File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/argparse.py", line 1367, in add_argument
>     return self._add_action(action)
>   File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/argparse.py", line 1730, in _add_action
>     self._optionals._add_action(action)
>   File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/argparse.py", line 1571, in _add_action
>     action = super(_ArgumentGroup, self)._add_action(action)
>   File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/argparse.py", line 1381, in _add_action
>     self._check_conflict(action)
>   File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/argparse.py", line 1520, in _check_conflict
>     conflict_handler(action, confl_optionals)
>   File "/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/lib/python3.7/argparse.py", line 1529, in _handle_conflict_error
>     raise ArgumentError(action, message % conflict_string)
> argparse.ArgumentError: argument -u/--username: conflicting option string: -u

I can change one option to an uppercase but then the long name has first character uppercase.

```Python
class Cli(Cliar):
    def _root(self, url="", Username=""):
        pass
```

> usage: cli.py [-h] [-u URL] [-U USERNAME] {} ...
> 
> optional arguments:
>   -h, --help            show this help message and exit
>   -u URL, --url URL
>   -U USERNAME, --**Username** USERNAME
> 
> commands:
>   {}                    Available commands:

Is is possible to produce:

>   -u URL, --url URL
>   -U USERNAME, --**username** USERNAME

I don't see a way with `set_metavars` either but perhaps I missed something in the docs.

**Comment by moigagoo** [comment]
Hi! Before I start, I am really, _really_ sorry for not responding for so long. I never understood GitHub's poor notifications feature, and haven't checked for new issues manually, so... Again, really sorry. It's all my fault, and I've set up email notifications for new issues already.

With regard to the reported issue. First, thank you so much for reporting it! Really appreciate it.

Second, there's no way to work around that as of Cliar 1.2.2, sadly. I think the easiest way to fix it would be to allow `set_arg_map` to accept a `Dict[str, tuple[str, str]]` as well as `Dict[str, str]`, which would allow to set uppercase short option names.

This way you could have fixed your code like so:
```python
from cliar import Cliar, set_arg_map


class Cli(Cliar):
    @set_arg_map({'username': ('username', '-U')})
    def _root(self, url="", username=""):
        pass


if __name__ == "__main__":
    cli = Cli()
    cli.parse()

```

**Comment by moigagoo** [comment]
Fixed in 1.2.3 with the new `set_sharg_map`. Your example can now be fixed:
```python
from cliar import Cliar, set_sharg_map

class Cli(Cliar):
    @set_sharg_map({"username": "U"})
    def _root(self, url="", username=""):
        pass


if __name__ == "__main__":
    cli = Cli()
    cli.parse()
```
